### PR TITLE
API endpoints for Media

### PR DESCRIPTION
**Pull request by sajjanstha**
### Task/Ticket

[API endpoints for Media](#)

#### The main changes are

- [x] Endpoint to save media
- [x] Endpoint to delete media

Note: Need to update Nodejs version to 20

### CODE REVIEW

File: prisma/migrations/20231130123148_create_media_table/migration.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+-- CreateTable
+CREATE TABLE "Media" (
+    "id" UUID NOT NULL DEFAULT gen_random_uuid(),
+    "url" TEXT NOT NULL,
+    "filename" TEXT NOT NULL,
+    "contentType" TEXT NOT NULL,
```
Comment (sazanrjb) [inline prisma/migrations/20231130123148_create_media_table/migration.sql:6]: is this extension or different like application/png?

**Review by sazanrjb — COMMENTED** [review]
We need to put mediaId in settings table for banner image.. Will this be done in another PR?

File: prisma/migrations/20231130123148_create_media_table/migration.sql (hunk 1 above)
Comment (sajjanstha) [reply prisma/migrations/20231130123148_create_media_table/migration.sql:6]: its mime like image/png, application/pdf, etc

**Comment by sazanrjb** [comment]
what do we store in bannerImage in settings then? it should be mediaId as it has all the metadata of the media, no?

File: prisma/schema.prisma [hunk 2]
```diff
@@ -13,11 +13,25 @@ datasource db {
 
 model Setting {
   id                String      @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
-  bannerImage       String?
+  mediaId           String?     @db.Uuid @unique
```
Comment (sazanrjb) [inline prisma/schema.prisma:16]: one last request.. Can this be called as bannerImageId and relation be called bannerImage? if we have to add other media in this table, names will be confusing..

Comment (sajjanstha) [reply prisma/schema.prisma]: done

**Review by sazanrjb — APPROVED** [review]

**Pull request merged**